11-Backbonejs/04-Backbone-Models.md

```diff
@@ -283,6 +283,23 @@ Good catch, hypothetical student! Ideally we shouldn't be calling `render()` in
 
 The reason is, if you're a view, you have no guarantee that all changes to the model will come through you. Maybe the model comes from a database server and is updated periodically. Maybe there are multiple views for the same model, and when one changes the others should pick it up. Model events can be an effective solution to all of these problems.
 
+### Re-Delegating DOM Events
+There's a bug in our application. Marking a task complete works fine, and adding a new task works fine, but once a new task has been added you can't mark _any_ task complete. Weird!
+
+The reason has to do with how DOM event listeners work. Remember that our `completeHandler` is bound to a click event on a specific button owned by our `TaskView`. But, look at `TaskListView.render()`. The first line reads
+
+```javascript
+this.listElement.empty();
```

Comment:
Not sure if it's worth bringing this in, but if we setup an event handler for the `add` event from our collection, which delegates to `addTask()` then we should be able to get away with:

```javascript
  createTask: function(event) {
    event.preventDefault();

    this.model.addTask(this.getInput());
    this.clearInput();
  }
```